src/libnnstreamer-edge/nnstreamer-edge-internal.c

```diff
@@ -812,8 +812,13 @@ _nns_edge_send_thread (void *thread_data)
   int64_t client_id;
   char *val;
   int ret;
+  uint queue_timeout_sec = 3000U;
 
-  while (nns_edge_queue_wait_pop (eh->send_queue, 0U, &data_h, &data_size)) {
+  if (NNS_EDGE_NODE_TYPE_PUB == eh->node_type)
+    queue_timeout_sec = 0U;
+
+  while (nns_edge_queue_wait_pop (eh->send_queue, queue_timeout_sec,
```

Comment:
Adding timeout here may cause an error if edge node sends new data after 3 seconds.
I suggest adding new function nns_edge_queue_stop () to stop and clear data in the queue.